### PR TITLE
implement a tasks endpoint

### PR DESCRIPTION
**Pull request by robrobbins**
still gatta write some specs, but it should look about like so...

### CODE REVIEW

File: apis/listing/listings.py [hunk 1]
```diff
@@ -92,7 +93,8 @@ def post(self):
             os.remove(loc)
 
             current_app.logger.info(C.NEW_LISTING_SUCCESS)
-            return {'message': C.NEW_LISTING_SUCCESS, 'task_id': uid}, 201
+
+            return dict(message=C.NEW_LISTING_SUCCESS, task_id=uid), 201
```
Comment (robrobbins) [inline apis/listing/listings.py:97]: been using this pattern everywhere else so...

File: apis/listing/tasks.py [hunk 2]
```diff
@@ -31,4 +31,7 @@ def send_data_hash_after_mining(tx_hash, listing, data_hash):
         # TODO timeout length?
         current_app.logger.info('Waiting for send data hash transaction receipt: %s', send_tx)
         send_rcpt = g.w3.eth.waitForTransactionReceipt(send_tx)
-        current_app.logger.info('Send data hash transaction mined: %s', send_rcpt['transactionHash'])
+        # return this as the task result as well. use hex...
+        send_hash = g.w3.toHex(send_rcpt['transactionHash'])
+        current_app.logger.info('Send data hash transaction mined: %s', send_hash)
+        return send_hash
```
Comment (robrobbins) [inline apis/listing/tasks.py:37]: returning a value from the task allows us to use `task.get` for a reason

**Comment by robrobbins** [comment]
#40

File: apis/task/tasks.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+api = Namespace('Tasks', description='Operations pertaining to celery asyncronous tasks')
+
+api.models['TaskResult'] = TaskResult
+
+@api.route('/<string:id>', methods=['GET'])
+class TaskRoute(Resource):
+    @api.expect(task_parser)
+    @api.marshal_with(TaskResult)
+    @api.response(200, C.CELERY_TASK_FETCHED)
+    @api.response(500, C.SERVER_ERROR)
```
Comment (rekibnikufesin) [inline apis/task/tasks.py]: is there any response from the backend that indicates an invalid or unknown task id was supplied? Or do we care?

File: tests/listing/test_listings.py [hunk 4]
```diff
@@ -155,8 +154,8 @@ def test_post_listings(mock_send, w3, voting, datatrust, listing, test_client, s
     data=test_payload)
 
     mock_send.assert_called_once_with(
-        HexBytes(tx).hex(),
-        HexBytes(listing_hash).hex(),
+        w3.toHex(tx),
```
Comment (rekibnikufesin) [inline tests/listing/test_listings.py:157]: 👍

**Review by rekibnikufesin — APPROVED** [review]

**Review by rekibnikufesin — APPROVED** [review]

**Pull request merged**